plugins/woocommerce/client/blocks/assets/js/base/stores/woocommerce/cart.ts

```diff
@@ -17,11 +17,18 @@ import type { Store as StoreNotices } from '@woocommerce/stores/store-notices';
  */
 import { triggerAddedToCartEvent } from './legacy-events';
 
+export type SelectedAttributesType = Omit< CartVariationItem, 'raw_attribute' >;
+
 export type OptimisticCartItem = {
 	key?: string;
 	id: number;
 	quantity: number;
 	variation?: CartVariationItem[];
+	type: string;
+};
+
+export type ClientCartItemType = Omit< OptimisticCartItem, 'variation' > & {
+	variation?: SelectedAttributesType[];
 };
```

Comment:
I would remove the `Type` part from the types' name. This would keep them shorter and more in line with the other types, which don't contain it either. So they would become `SelectedAttributes` and `ClientCartItem`. How does it sound?